### ISSUE TITLE
 Incorrect Test Case Matching with Double Quotes

### ISSUE THREAD

**Issue by Anker101** [damies13/rfswarm#159]
**Describe the bug**
When executing test cases with double quotes in the test case name using the Robot Framework, the test cases are not matched correctly, resulting in an error.

**To Reproduce**

Create a test case with double quotes in the name, like:

Test Case    Send GET on API "${endpoint}" on "${env}" for "${tenant}" should be 200

Execute the following command:

$ robot -t "Send GET on API "\${endpoint}" on "\${env}" for "\${tenant}" should be 200"  "Anker01_variables.robot"

The test case is not matched correctly, and the output is as follows:

[ ERROR ] Suite 'Anker01 variables' contains no tests matching name 'Send GET on API ${endpoint} on ${env} for ${tenant} should be 200'.

**Expected behavior**
The test case should be matched and executed successfully.

**Additional context**
The issue seems to be related to how the Robot Framework handles escaped double quotes in the command for specifying test cases with double quotes in their names. Manually including escaped double quotes in the command allows the test case to be matched correctly and executed. 
$ robot -t "Send GET on API \"\${endpoint}\" on \"\${env}\" for \"\${tenant}\" should be 200"  "Anker01_variables.robot"

This correctly identifies the test case and executes it, as seen in the output:

Send GET on API "my endpoint" on "QAENV" for "Anker02" should be 200  | PASS |

This bug affects users who need to work with test case names containing double quotes in their projects.

**Comment by damies13** [comment]
Info from chat on Discord

------------------------------------------------------------------------------------------------------------------------------------------------------
Anker01 — 02/08/2023 18:25
------------------------------------------------------------------------------------------------------------------------------------------------------

Hello ,
The log says : [ ERROR ] Suite 'API' contains no tests matching name 'Send GET on API ${endpoint} on ${env} for ${tenant} should be 200'.
But in my scenario i specified that the name of the test case is : 
Send GET on API "${endpoint}" on "${env}" for "${tenant}" should be 200 
with quotation marks

------------------------------------------------------------------------------------------------------------------------------------------------------
damies13 — 02/08/2023 19:13
------------------------------------------------------------------------------------------------------------------------------------------------------

Thanks, I'll try to reproduce that

------------------------------------------------------------------------------------------------------------------------------------------------------
damies13 — 02/08/2023 19:30
------------------------------------------------------------------------------------------------------------------------------------------------------

OK found the issue, looks like I need to escape the double quotes as well
```
$ robot -t "Send GET on API "\${endpoint}" on "\${env}" for "\${tenant}" should be 200"  "Anker01_variables.robot"
[ ERROR ] Suite 'Anker01 variables' contains no tests matching name 'Send GET on API ${endpoint} on ${env} for ${tenant} should be 200'.
```
Try --help for usage information.

vs
```
$ robot -t "Send GET on API \"\${endpoint}\" on \"\${env}\" for \"\${tenant}\" should be 200"  "Anker01_variables.robot"
==============================================================================
Anker01 variables                                                             
==============================================================================
Send GET on API "my endpoint" on "QAENV" for "Anker02" should be 200  | PASS |
------------------------------------------------------------------------------
Anker01 variables                                                     | PASS |
1 test, 1 passed, 0 failed
==============================================================================
```
------------------------------------------------------------------------------------------------------------------------------------------------------

**Comment by damies13** [comment]
Confirmed the Double Quotes also need to be escaped on Window, but as I expected the `${` shouldn't be escaped on Windows

**Comment by damies13** [comment]
Merged into release branch v1.1.5
